### ISSUE TITLE
CMake error when running `build-script` with `--xcode`

### ISSUE THREAD

**Issue by whiteio** [apple/swift#62645]
**Description**

Build script fails on running:
```
utils/build-script --skip-build-benchmarks --skip-ios \
   --skip-watchos --skip-tvos --swift-darwin-supported-archs "$(uname -m)" \
   --swift-disable-dead-stripping --release-debuginfo --xcode --clean
```

This issue appears to be similar to the issue discussed in #62205

**Steps to reproduce**

1. Checkout main: `utils/update-checkout --scheme main` (current revision: 244ca4e2426260e7b9161c2fd6534dc350983cdf)
2. Run the build script using the following command:
```
utils/build-script --skip-build-benchmarks --skip-ios \
   --skip-watchos --skip-tvos --swift-darwin-supported-archs "$(uname -m)" \
   --swift-disable-dead-stripping --release-debuginfo --xcode --clean
```

There following errors occur, 4 times each:

```
CMake Error in lib/ASTGen/CMakeLists.txt:
  Imported target "SwiftSyntax::SwiftBasicFormat" includes non-existent path

    "/Users/whiteio/Development/swift-project/build/Xcode-RelWithDebInfoAssert/earlyswiftsyntax-macosx-arm64/lib/swift/host"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
```

```
CMake Error in lib/Parse/CMakeLists.txt:
  Imported target "SwiftSyntax::SwiftBasicFormat" includes non-existent path

    "/Users/whiteio/Development/swift-project/build/Xcode-RelWithDebInfoAssert/earlyswiftsyntax-macosx-arm64/lib/swift/host"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
```

**Expected behavior**

It should successfully build and create the Xcode projects.

**Environment**

swift-driver version: 1.62.15 
Apple Swift version 5.7.1 (swiftlang-5.7.1.135.3 clang-1400.0.29.51)
Target: arm64-apple-macosx12.0

Xcode 14.1
Build version 14B47b

**Comment by michaeleisel** [comment]
I see this behavior as well

**Comment by LucianoPAlmeida** [comment]
Ok I just faced this as well, what happens is that this is a failure is a dependency package that is required when building is missing. Dependencies are fetched when we run `./utils/update-checkout --clone` so if new dependency paths are introduced we would need to run it again to force the fetch of recent introduced things. 
TL;DR : Run `./utils/update-checkout --clone` and re-run `build-script` ideally with `--clean`. 
This should fix the issue

**Comment by AnthonyLatsis** [comment]
@LucianoPAlmeida Did this actually work for you? I don‘t see any new repos in the update-checkout config either.

**Comment by AnthonyLatsis** [comment]
@DougGregor Could you take a look at this please? I think it was #62326. I have no idea what exactly broke the build, but we probably should make `--xcode` imply `--skip-early-swiftsyntax` in addition to `--skip-early-swift-driver`, since we can use their respective package files to work in Xcode instead.

**Comment by LucianoPAlmeida** [comment]
> @LucianoPAlmeida Did this actually work for you? I don‘t see any new repos in the update-checkout config either.

Yes, when I saw the error the idea was update dependencies with new changes either existing dependencies or  fetch new added if there are any.  
Maybe `--clone` wasn't needed and the fix is just to fetch new changes in existing repos... but I had in my mind that `--clone` force fetch updates on existing repos so I always go with it when updating checkout.

**Comment by whiteio** [comment]
> Ok I just faced this as well, what happens is that this is a failure is a dependency package that is required when building is missing. Dependencies are fetched when we run `./utils/update-checkout --clone` so if new dependency paths are introduced we would need to run it again to force the fetch of recent introduced things. TL;DR : Run `./utils/update-checkout --clone` and re-run `build-script` ideally with `--clean`. This should fix the issue

This didn't resolve the issue for me.

**Comment by MikePendo** [comment]
I have run the build with `--clean` and still had it

**Comment by AnthonyLatsis** [comment]
Workaround: pass `--skip-early-swiftsyntax`.
